docs/source/api.rst

```diff
@@ -536,6 +536,42 @@ retworkx API
    :return matrix: The adjacency matrix for the input dag as a numpy array
    :rtype: numpy.ndarray
 
+.. py:function:: graph_all_simple_paths(graph, from, to, min_depth=None, cutoff=None)
+   Return all simple paths between 2 nodes in a PyGraph object
+
+   A simple path is a path with no repeated nodes.
+
+   :param PyGraph graph: The graph to find the path in
+   :param int from: The node index to find the paths from
+   :param int to: The node index to find the paths to
+   :param int min_depth: The minimum depth of the path to include in the output
+       list of paths. By default all paths are included regardless of depth,
+       sett to 0 will behave like the default.
+   :param int cutoff: The maximum depth of path to include in the output list
+       of paths. By default includes all paths regardless of depth, setting to
+       0 will behave like default.
+
+   :returns paths: A list of lists where each inner list is a path path
+   :rtype: list
+
+.. py:function:: dag_all_simple_paths(graph, from, to, min_depth=None, cutoff=None)
```

Comment:
In a DAG, aren't all paths simple? Should this just be `all_paths`, or `dag_all_paths`?

Reply:
Yeah in a true dag they are, but with the cycle check disabled PyDAG could be just used as a directional graph, and all the functions that require a DAG have a cycle check in them. What I've been meaning to do is make `PyDAG` a subclass of a new class `PyDiGraph` and remove the entire cycle checking bit from `PyDiGraph` (see #20). The problem is subclassing in rust/PyO3 is a bit weird and I haven't been able to find a working pattern for it yet. It's partially because rust doesn't actually do OO like python, polymorphism is done via shared traits between types and I'm too inexperienced with rust an pyo3 to work my way out of that conflict sp far. But until that's ready I'm just treating PyDAG as a directed graph here since unless cycle check is set it really is one.

Reply:
I pushed up https://github.com/Qiskit/retworkx/pull/88 to try and resolve the confusing api around this.